config/filter.d/zoneminder.conf

```diff
@@ -0,0 +1,21 @@
+# Fail2Ban filter for Zoneminder login failures
+
+[Definition]
+
+# pattern: [Wed Apr 27 23:12:07.736196 2016] [:error] [pid 2460] [client 10.1.1.1:47296] WAR [Login denied for user "test"], referer: https://zoneminderurl/index.php
+#
+#
+# Option:  failregex
+# Notes.:  regex to match the password failure messages in the logfile. The
+#          host must be matched by a group named "host". The tag "<HOST>" can
+#          be used for standard IP/hostname matching and is only an alias for
+#          (?:::f{4,6}:)?(?P<host>[\w\-.^_]+)
+
+failregex = ^\[\](?: \[:error\])?(?: \[pid \d+\])? \[client <HOST>:\d+\] WAR \[Login denied for user "[^"]*"\]
```

Comment:
I would have added optional entry to match referer and anchor at the end

Reply:
I'm sure @yarikoptic said as "optional entry" something like this:
```regex
(?:, referer: \S*)$
```
You've added it as second regex (that in addition will never match, because the first one always wins)...
Since the part before `<HOST>` is not variable I don't see any reason for this referer part. IMHO the message `Login denied for user` is precise enough no matter what for data follows this at end of line.

I'll merge it without second regex. Thanks.